polygeist/tools/cgeist/Lib/CodeGenTypes.h

```diff
@@ -95,6 +96,7 @@ class CodeGenTypes {
                               llvm::StructType *ST);
 
   clang::QualType getPromotionType(clang::QualType Ty) const;
+  clang::CanQualType getCanonicalType(clang::QualType T) const;
```

Comment:
In clang-mlir.cc we can access the `getCanonicalType` member function of ASTContext so we do not need this one. See comment I placed in clang-mlir.cc.